diff --git a/src/main/java/soot/SootResolver.java b/src/main/java/soot/SootResolver.java
index a9cddbaa2..7568cab69 100644
--- a/src/main/java/soot/SootResolver.java
+++ b/src/main/java/soot/SootResolver.java
@@ -30,6 +30,7 @@ import java.util.Collection;
 import java.util.Deque;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -398,4 +399,9 @@ public class SootResolver {
       super(s);
     }
   }
+
+  public Set<soot.Type> getReferenceSignatures(SootClass sc){
+    return this.classToTypesSignature.get(sc);
+  }
+
 }

